composeApp/src/commonMain/kotlin/design-system/ProfileIcon.kt

```diff
@@ -0,0 +1,80 @@
+package design
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import common.conditional
+
+object ProfileIconConfig {
+    enum class Context {
+        PreLogin, Stories;
+
+        fun shouldHighlight(): Boolean {
+            return when(this) {
+                PreLogin -> false
+                Stories -> true
+            }
+        }
+
+        fun usernameShouldBeBold(): Boolean {
+            return when(this) {
+                PreLogin -> true
+                Stories -> false
+            }
+        }
+    }
```

Comment:
This is usually a bad practice, this can be replaced with constant values, without requiring calculating it.

```suggestion
    enum class Context(
        val shouldHighlight: Boolean,
        val usernameShouldBeBold: Boolean,
    ) {
        PreLogin(
          shouldHighlight = false,
          usernameShouldBeBold = true,
        ),
        Stories(
          shouldHighlight = true,
          usernameShouldBeBold = false,
        )
    }
```

This approach here is also more [Stable](https://developer.android.com/jetpack/compose/performance/stability) for Compose it self, because a non Composable function call inside a Composable is always unstable and will always rerun the calculation and recompose

Reply:
Interesting, I didn't know we could define read-write values for enums.